rover_auxiliary/src/ipcamera_api.py

```diff
@@ -0,0 +1,322 @@
+
+    def setBrightness(self, value: int):
+        self.set_onvif_param('Brightness', value)
+
+    def setContrast(self, value: int):
+        self.set_onvif_param('Contrast', value)
+    
+    def setSaturation(self, value: int):
+        self.set_onvif_param('ColorSaturation', value)
+    
+    def setSharpness(self, value: int):
+        self.set_onvif_param('Sharpness', value)
+    
+    def setResolution(self, value: string):
+        if value in self.RESOLUTION_MAP:
+            width, height = self.RESOLUTION_MAP[value]
+            extra = (width, height)
+        else:
+            self.logger.error(f"Invalid resolution value: {value}")
+            return False
+
+        self.set_onvif_param('Resolution', value, extra)
+
+    def setFrameRate(self, value: int):
+        self.set_onvif_param('FrameRateLimit', value)
+    
+    def setBitrate(self, value: int):
+        self.set_onvif_param('BitrateLimit', value)
+    
+    def disableWideDynamicRange(self):
+        self.set_http_param('WideDynamicRange', '0')
+    
+    def setWideDynamicRangeLevel(self, value: int = 50):
+        self.set_http_param('WideDynamicRange', '1')
+        self.set_http_param('WideDynamicRangeLevel', value)
+    
+    def enableBackLight(self):
+        self.set_http_param('BackLight', '1')
+    
+    def disableBackLight(self):
+        self.set_http_param('BackLight', '0')
+    
+    def HorizontalMirror(self):
+        self.set_http_param('MirrorHorizontal', '1')
+    
+    def VerticalMirror(self):
+        self.set_http_param('MirrorVertical', '1')
+    
+    def resetHorizontalMirror(self):
+        self.set_http_param('MirrorHorizontal', '0')
+    
+    def resetVerticalMirror(self):
+        self.set_http_param('MirrorVertical', '0')
+
+    def enableAntiFalseColor(self):
+        self.set_http_param('AntiFalseColor', '1')
+    
+    def disableAntiFalseColor(self):
+        self.set_http_param('AntiFalseColor', '0')
+    
+    def enableDigitalImageStabilizer(self):
+        self.set_http_param('DigitalImageStabilizer', '1')
+    
+    def disableDigitalImageStabilizer(self):
+        self.set_http_param('DigitalImageStabilizer', '0')
+    
+    def enableLensShadeCorrection(self):
+        self.set_http_param('LensShadeCorrection', '1')
+    
+    def disableLensShadeCorrection(self):
+        self.set_http_param('LensShadeCorrection', '0')
+    
+    def setLensDistortionCorrection(self, value: int = 50):
+        self.set_http_param('LensDistortionCorrection', '1')
+        self.set_http_param('LensDistortionCorrectionLevel', value)
+    
+    def disableLensDistortionCorrection(self):
+        self.set_http_param('LensDistortionCorrection', '0')
+    
+    def setAntiFog(self, value: int = 50):
+        self.set_http_param('AntiFog', '1')
+        self.set_http_param('AntiFogLevel', value)
+    
+    def disableAntiFog(self):
+        self.set_http_param('AntiFog', '0')
+    
+    def setScene(self, value = 'indoor'):
```

Comment:
Can we use a map/enum, that way we don't have to check the source code to know valid inputs? (indoor, outdoor, etc.)